charts/argo-templates/templates/_helpers.tpl

```diff
@@ -32,3 +32,19 @@ retryStrategy:
 {{ . }}
 {{- end }}
 {{- end }}
+
+{{/* Collect metrics about execution results, including duration and result statuses. */}}
+{{- define "argo.collect-execution-metrics" }}
+- name: result_counter
+  help: "Count step executions by result status"
+  labels:
+    - key: status
+      value: "{{status}}"
+  counter: 
+    value: "1"
+- name: realtime_duration_gauge
```

Comment:
```suggestion
- name: realtime_duration
```
Similar reasoning as above: The fact that it's a "gauge" is an artifact of Prometheus. The data itself is just a duration.